Source/ARTRealtimeChannel.m

```diff
@@ -989,9 +989,12 @@ - (void)setSuspended:(ARTStatus *)status retryIn:(NSTimeInterval)retryTimeout {
     [self failQueuedMessages:status];
     [self transition:ARTRealtimeChannelSuspended status:status];
     [[self unlessStateChangesBefore:retryTimeout do:^{
+        [self.realtime.logger debug:__FILE__ line:__LINE__ message:@"RT:%p C:%p (%@) reattach initiated by retry timeout", self->_realtime, self, self.name];
         [self reattachWithReason:nil callback:^(ARTErrorInfo *errorInfo) {
-            ARTStatus *status = [ARTStatus state:ARTStateError info:errorInfo];
-            [self setSuspended:status];
+            if (errorInfo) {
```

Comment:
> unlessStateChangesBefore:retryTimeout do

Why are we still setting the retry timeout unconditionally when we go into suspended?

In particular, if a channel goes into suspended as a side effect of a connection state change, it should not have any kind of independent retry timeout of its own volition; it should just sit there doing nothing until the connection state becomes CONNECTED, at which time it should have the side effect of triggering all `suspended` channels to reattach (RTL3d). 

The independent channel suspended retry timer from [RTL13](https://docs.ably.io/client-lib-development-guide/features/#RTL13) is only for if the channel refuses to become attached _while the connection is connected_, per rtl4f. E.g. cf https://github.com/ably/ably-js/blob/master/common/lib/client/realtimechannel.js#L414

Reply:
@SimonWoolf Fixed with 45c21e4.